src/main/java/org/truffleruby/core/numeric/IntegerNodes.java

```diff
@@ -1956,7 +1965,7 @@ protected Object upto(long from, long to, RubyProc block) {
 
         @Specialization
         protected Object upto(long from, double to, RubyProc block) {
-            return upto(from, (long) Math.ceil(to), block);
+            return upto(from, (long) Math.floor(to), block);
```

Comment:
Is this change intentional?

Reply:
Because the `(int, double)` specialization uses floor() too, I see.

Reply:
Right - spec for `#upto` was failing.

Reply:
Aside -- I wonder if `(long) Math.floor(to)` is the same as `(long) to` given `to` is a `double` (same for `int` instead of `long`).

Reply:
If it is, I'll let the compiler work that out and do it for me!